README.md

```diff
@@ -51,29 +86,66 @@ jobs:
     #
     # Default: main
     main-branch-name: ''
 
-    # The glob(7) pattern to be provided to `git describe --match` in order to match against
-    # the latest relevant tag on the specified "main" branch.
-    #
-    # The default pattern aligns with the default behavior of the complementary `nrwl/nx-tag-successful-ci-run` action.
-    #
-    # Default: nx_successful_ci_run*
-    tag-match-pattern: ''
-
     # Applies the derived SHAs for base and head as NX_BASE and NX_HEAD environment variables within the current Job.
     #
     # Default: true
     set-environment-variables-for-job: ''
 
-    # By default, if no matching tags are found on the main branch to determine the SHA, we will log a warning and use HEAD~1. Enable this option to error and exit instead.
+    # By default, if no successful workflow run is found on the main branch to determine the SHA, we will log a warning and use HEAD~1. Enable this option to error and exit instead.
     #
     # Default: false
-    error-on-no-matching-tags: ''
+    error-on-no-successful-workflow: ''
+
+    # The ID of the github action workflow to check for successful run or the name of the file name containing the workflow. 
+    # E.g. 'ci.yml'. If not provided, current workflow id will be used
+    #
+    workflow-id: ''
 ```
-<!-- end configuration-options -->
\ No newline at end of file
+<!-- end configuration-options -->
+
+## Background
+
+When we run `affected` command on [Nx](https://nx.dev/), we can specify 2 git history positions - base and head, and it calculates [which projects in your repository changed
+between those 2 commits](https://nx.dev/latest/angular/tutorial/11-test-affected-projects#step-11-test-affected-projects
+). We can then run a set of tasks (like building or linting) only on those **affected** projects.
+
+This makes it easy to set-up a CI system that scales well with the continous growth of your repository, as you add more and more projects.
+
+
+### Problem
+
+Figuring out what these two git commits are might not be as simple as it seems.
+
+On a CI system that runs on submitted PRs, we determine what commits to include in the **affected** calculation by comparing our `HEAD-commit-of-PR-branch`to the commit in main branch (`master` or `main` usually) from which the PR branch originated. This will ensure our PR in whole is always being tested.
+
+But what if we want to set up a continuous deployment system
+that, as changes get pushed to `master`, it builds and deploys
+only the affected projects?
+
+What are the `FROM` and `TO` commits in that case?
+
+They can't be just `HEAD` and `HEAD~1` as some of those run might fail. If a few deployments fail one after another, that means that we're accumulating a list of affected projects that are not getting deployed. Anytime we retry the deployment, we want to include **every commit since the last time we deployed successfully**. That way we ensure we don't accidentally skip deploying a project that has changed.
```

Comment:
```suggestion
Conceptually, what we want is to use the absolute latest commit on the `master` branch as the HEAD and the previous _successful_ commit on `master` as the BASE. Note, we want the previous _successful_ one because it is still possible for commits on the `master` branch to fail for a variety of reasons.

The commits therefore can't just be `HEAD` and `HEAD~1`. If a few deployments fail one after another, that means that we're accumulating a list of affected projects that are not getting deployed. Anytime we retry the deployment, we want to include **every commit since the last time we deployed successfully**. That way we ensure we don't accidentally skip deploying a project that has changed.
```